src/modules/tasks/components/TaskForm/FormAddAndEdit.tsx

```diff
@@ -0,0 +1,164 @@
+import { Link, useNavigate, useParams } from "react-router-dom";
+import styles from "./CSS/TaskForm.module.css";
+import { useForm } from "react-hook-form";
+import { axiosInstance, PROJECT_URL, TASKSURL, USERSURL } from "../../../../services/EndPoints";
+import { useEffect, useState } from "react";
+import { toast } from "react-toastify";
+
+
+interface AddTasks {
+  title: string,
+  description: string,
+  employeeId: string | number,
+  projectId: string | number,
+}
+
+const FormAddAndEdit = () => {
+  const [projects, setProjects] = useState();
+  const [users, setUsers] = useState();
+  const {register, handleSubmit, setValue,formState:{isSubmitting, errors}} = useForm();
+  const params = useParams()
+  const tasksId = params.taskId
+  const isTasks = tasksId == "create-task"
+  const navigate = useNavigate()
+
+  const onSumbit = async(data:AddTasks)=>{
+    try {
+      await axiosInstance[isTasks? "post" : "put"]( isTasks? TASKSURL.POST_TASK 
+        : TASKSURL.PUT_TASK(tasksId),data
+      )
+      toast.success("Add Tasks")
+      navigate("/tasks")
+
+    } catch (error) {
+      console.log(error);
+      
+    }
+  }
+  const getprojects = async () => {
+    const res = await axiosInstance.get(PROJECT_URL.GET_PROJECTS)
+    console.log(res.data.data);
+    
+    setProjects(res.data.data)
+  }
+  const getUsers = async () => {
```

Comment:
We will need to set a large number like 1000, and make the select for the users searchable to be able to choose the employee we want. The current implementation will show only 5.